### PR TITLE
refactor(ui): move onboarding screen to TypeScript

### PR DESCRIPTION
**Pull request by geigerzaehler**

### CODE REVIEW

File: ui/Screen/Onboarding/EnterPassphrase.svelte [hunk 1]
```diff
@@ -61,14 +67,16 @@
...
     passphraseValidation = getValidationState("passphrase", validations);
     repeatedPassphraseValidation = getValidationState(
       "repeatedPassphrase",
+      // @ts-expect-error: `validations` is guaranteed to be a validations object
       validations
     );
   };
 
-  $: validate(passphrase, repeatedPassphrase);
+  $: validate();
```
Comment (rudolfs) [inline ui/Screen/Onboarding/EnterPassphrase.svelte:79]: Is this a typo?

Comment (rudolfs) [reply ui/Screen/Onboarding/EnterPassphrase.svelte:79]: Because if you don't list the variables the reactive statement won't trigger.

Comment (geigerzaehler) [reply ui/Screen/Onboarding/EnterPassphrase.svelte:79]: `validate` does not have any arguments. The original code was an oversight.

Comment (rudolfs) [reply ui/Screen/Onboarding/EnterPassphrase.svelte:79]: I guess we're just abusing the parameters. Not sure it'll work, but maybe we can rewrite it to something like this, to make it more explicit?
```
$: passphrase; repeatedPassphrase; validate();
```

Comment (rudolfs) [reply ui/Screen/Onboarding/EnterPassphrase.svelte]: 3bedfee09431dd6318216b48882c2c8a2d681122 -> this also works 👍

**Pull request merged**